### DISCUSSION TITLE
not supported resolving ES modules imported from build

### DISCUSSION THREAD

**Discussion by mbkProgrammer** [remix-run/remix#7586] [Q&A] [answered]
After I update remix from 1.19.3 to 2.0.1 when I want serve build I have this error 
```
Error: Directory import 'C:\Users\mbk20\Projects\pertadex\build' is not supported resolving ES modules imported from C:\Users\mbk20\Projects\pertadex\node_modules\@remix-run\serve\dist\cli.js
Did you mean to import C:/Users/mbk20/Projects/pertadex/build/index.js?
    at new NodeError (node:internal/errors:400:5)
    at finalizeResolution (node:internal/modules/esm/resolve:319:17)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:842:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ESMLoader.import (node:internal/modules/esm/loader:525:22)
    at importModuleDynamically (node:internal/modules/cjs/loader:1151:29)
    at importModuleDynamicallyWrapper (node:internal/vm/module:438:21)
```
this is my remix.config.js file:
```
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  ignoredRouteFiles: ['**/.*'],
  tailwind: true,
  postcss: true,
  // appDirectory: "app",
  // assetsBuildDirectory: "public/build",
  // serverBuildPath: "build/index.js",
  // publicPath: "/build/",
  serverModuleFormat: 'cjs',

  future: {},
  serverDependenciesToBundle: [
    'swiper',
    'swiper/react',
    'swiper/react/swiper-react.js',
    'ssr-window',
    'ssr-window/ssr-window.esm.js',
    'dom7',
  ],
};
```

**Comment by NinthAutumn** [comment, marked as answer]
for anyone currently experiencing this problem i fixed it by changing the package.json:
 "start":"remix-serve build" 
 to
 "start":"remix-serve ./build/index.js"

**Reply by RoniqueRicketts** [reply]
I am experiencing this error in version 2.2.0, Here is my scripts in package.json
```
"scripts": {
    "build": "remix build",
    "dev": "remix dev --manual -c \"node --watch-path server.js --watch server.js\"",
    "start": "cross-env NODE_ENV=production node ./server.js",
    "typecheck": "tsc"
  },
  ```

**Reply by RoniqueRicketts** [reply]
I am just going to keep this here but I added ```--experimental-specifier-resolution=node``` in the start script and it solved the problem. I was importing Material UI to the project when I got this error so if anyone else got this error you could use the above flag.

**Reply by alexey-sh** [reply]
![image](https://github.com/remix-run/remix/assets/1584156/103706aa-5a43-4764-ba6c-d57b4e6dc911)

**Reply by rahad06** [reply]
this is how it was by default and it did not stop my app from erroring:
 "start":"remix-serve ./build/index.js"
![image](https://github.com/remix-run/remix/assets/97092525/d523dabf-1c66-4f3e-a85e-1fd3c9520d6d)
![image](https://github.com/remix-run/remix/assets/97092525/143ce48e-a5b0-425b-b94a-e10b7878aa51)
i even have the server config for it which should stop the error but it doesnt.

**Comment by mkromann** [comment]
I am running into the same issue with 2.1. I do not understand it though but Ninth's fix did it for me. If somebody can explain why I would be grateful.

**Comment by DKeen0123** [comment]
I also experienced this moving to v2. Ninth's fix does seem to work for me too. Looking at the discord, a bunch of folks upgrading have run into this one. Would be a good one for the Remix team to address

**Comment by hanqitech** [comment]
https://github.com/mui/material-ui/tree/master/examples/material-ui-remix-ts
I solved the problem according to this ,hope this could help you .

**Comment by Ikdemm** [comment]
I am facing the same problem after migrating to `v2.3.0`
The solution of changing the start script to `"start":"remix-serve ./build/index.js"` worked for me :pray:

**Reply by zolzaya** [reply]
Worked for me :)

**Comment by nadiTime** [comment]
How would you solve this for dev environment?

**Comment by observerkei** [comment]
Does anyone know how to solve this error when SPA mode works (no server)? I ran into this problem when trying to run material templates.
